lib/reversetunnel/remotesite.go

```diff
@@ -790,8 +790,8 @@ func (s *remoteSite) DialTCP(params reversetunnelclient.DialParams) (net.Conn, e
 }
 
 func (s *remoteSite) dialAndForward(params reversetunnelclient.DialParams) (_ net.Conn, retErr error) {
-	if params.GetUserAgent == nil && params.AgentlessSigner == nil {
-		return nil, trace.BadParameter("user agent getter and agentless signer both missing")
+	if params.GetUserAgent == nil && !params.IsAgentlessNode {
+		return nil, trace.BadParameter("user agent getter is required for non-agentless nodes")
```

Comment:
"non-agentless" is a double negation, what would you say for this:

```suggestion
		return nil, trace.BadParameter("the node is not agentless, but the user agent getter is missing")
```

Reply:
I struggled to find a good terminology here
"not agentless" is also double negation, right?

What do you think about "user agent getter is required for teleport nodes"?